From ffa8d43e6db3d657ab69286db2c1b290074647ea Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Mon, 24 Aug 2015 17:59:16 +0800
Subject: [PATCH] hdmi:rk3288/rk3368: support lte_340mcsc_scramble.

If EDID indicate sink support lte_340mcsc_scramble, we enable scrambing mode.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 .../video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
index 890d08266582..ae940159c670 100755
--- a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
@@ -590,7 +590,9 @@ static int rockchip_hdmiv2_video_framecomposer(struct hdmi *hdmi_drv,
 	hdmi_msk_reg(hdmi_dev, FC_INVIDCONF,
 		     m_FC_HDCP_KEEPOUT, v_FC_HDCP_KEEPOUT(1));
 	if (hdmi_drv->edid.scdc_present == 1) {
-		if (tmdsclk > 340000000) {/* used for HDMI 2.0 TX */
+		if (tmdsclk > 340000000 ||
+		    hdmi_drv->edid.lte_340mcsc_scramble) {
+			/* used for HDMI 2.0 TX */
 			mutex_lock(&hdmi_dev->ddc_lock);
 			rockchip_hdmiv2_scdc_init(hdmi_dev);
 			sink_version =
-- 
2.35.3

